app/components/shared/process_steps_component.rb

```diff
@@ -0,0 +1,31 @@
+class Shared::ProcessStepsComponent < ViewComponent::Base
+  def initialize(process:, service:)
+    @process = process
+    @service = service
+  end
+
+  def render?
+    @process.blank? || %w[create review].include?(@process.state)
+  end
+
+  def current_step_number
+    @service.current_step_number
+  end
+
+  def steps_to_display
+    @service.steps_to_display
+  end
+
+  def total_steps
+    @service.total_steps
+  end
+
+  def set_active_step_class(step_number, current_step_number)
```

Comment:
These two can probably just be `active_step_class` and `visited_step_class`.

I'm on a crusade against the word "set" today 😂